core/src/test/java/novoda/lib/sqliteprovider/provider/SQLiteProviderLocalTest.java

```diff
@@ -212,10 +281,24 @@ private void insert(String uri, ContentValues cv) {
         provider.insert(Uri.parse("content://" + uri), cv);
     }
 
+    private void bulkInsert(String uri, ContentValues[] cv) {
+        provider.bulkInsert(Uri.parse("content://" + uri), cv);
+    }
+
     private void query(String uri) {
         provider.query(Uri.parse("content://" + uri), null, null, null, null);
     }
 
+    private ContentValues[] createContentValuesArray(int size) {
+        ContentValues[] bulkToInsert = new ContentValues[size];
+        ContentValues contentValues = new ContentValues();
+        contentValues.put("test", "test");
+        for (int i = 0; i < size; i++) {
+            bulkToInsert[i] = contentValues;
+        }
+        return bulkToInsert;
+    }
+
     public class SQLiteProviderImpl extends SQLiteContentProviderImpl {
```

Comment:
Hmm, I'm definitely a classist TDDer, as it is more robust IMO, and makes for easier refactoring. I'll persuade you guys yet :tongue: I like testing the input and asserting the output though, even if we are overriding for testing. 

Also if delete returns 0 (line 251), doesn't that mean no deletes have happened and you won't want to notify? I think I'm missing something here.